### DISCUSSION TITLE
[Unity] Timeout exception

### DISCUSSION THREAD

**Discussion by iulianbroasca** [WalletConnect/walletconnect-monorepo#2633] [Developer Support]
Hey, 

After importing the DLLs in Unity, I took the example, adjusted with the projectId and MetaData for my project and I tested it. On UnityEditor I'm able to init and connect, but on mobile I got TimeOutException.

For 2.0.4 Exception: TimeoutException: Unavailable WS RPC url at WalletConnectSharp.Common.Utils.Extensions.WithTimeout
For 2.0.5 Exception: TimeoutException: socket stalled at WalletConnectSharp.Common.Utils.Extensions.WithTimeout

Has anyone else encountered this problem?

**Comment by finessevanes** [comment]
@iulianbroasca can you please include the configuration code, SDK version, and repro steps? If you could include a screenshot/video of the error that would be great!

**Reply by iulianbroasca** [reply]
Hi,

I took the code from execute [method](https://github.com/WalletConnect/WalletConnectSharp/blob/2.0/Tests/WalletConnectSharp.Examples/SimpleExample.cs) and I adjusted the project Id and Metadata for us and I tested on both situation. On editor everything works fine, on mobile I got exceptions. We used 2.0.5 .

**Reply by iulianbroasca** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/37576450/64a1aa2c-1c2b-473e-afd3-d1581919ab9d)

**Reply by iulianbroasca** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/37576450/2f97ec2c-a6d9-47f5-9972-bff0c22994b1)

**Reply by iulianbroasca** [reply]
We got the exceptions after calling init method.

**Reply by iulianbroasca** [reply]
Hi @finessevanes,

Do you have some updates? :-) Should I provide more information?

Thank you :-)

**Reply by mustafadervisglu** [reply]
@finessevanes Everyone is getting this error in unity. can you help me please?

**Reply by iulianbroasca** [reply]
Hey @mustafadervisglu,

Try this repo https://github.com/WalletConnect/WalletConnectUnity . On the 2.0 branch, you can find the WCv2 integration for Unity. The sample works fine but it's still in development.
